module3/lessons/many-to-many.md

```diff
@@ -0,0 +1,608 @@
+---
+title: Many to Many Relationships
+tags: database, schema, relationships
+---
+
+## Learning Goals
+
+- Describe the relationship between a foreign key on one table and a primary key on another table
+- Diagram a one-to-many relationship
+- Diagram a many-to-many relationship
+- Understand what a join table is and why we need one
+- Create many-to-many relationships in Rails
+- Interpret `schema.rb`
+
+## Vocabulary
+
+- Foreign Key
+- Primary Key
+- One-to-Many Relationship
+- Many-to-Many Relationship
+- Join Table
+- Migration
+- Schema
+
+## Setup
+
+TODO: update starter repo branch for song tests with song serializer
+
+You can begin this lesson on the `many-to-many-setup` branch of the Set List Tutorial Repo [here](https://github.com/turingschool-examples/set-list-api/tree/many-to-many-setup).
+
+## Warm-Up
+
+In SetList:
+
+1. What is the relationship between Songs and Artists?
+2. Diagram the database tables for Songs and Artists.
+
+# One-to-Many Relationships
+
+- **Primary Key** - a key in a relational database that is unique for each record. This is also known as an `id`. It is a unique identifier, such as a driver's license, or the VIN on a car. You may have one and only one primary key value per table.
+- **Foreign Key** - a foreign key is a field in one table that uniquely identifies a row of another table. A foreign key is defined in a second table, but it refers to the primary key in the first table.
+- The relationship between `songs` and `artists` is a one-to-many relationship.
+- `songs` has a column called `artist_id` which refers to the primary key of the `artist` table.
+- `artist_id` is the foreign key
+
+**Songs Table**
+
+| id | title | length | play_count | artist_id |
+| --- | --- | --- | --- | --- |
+| 1 | This Must Be the Place | 345 | 23 | 1 |
+| 2 | Aint No Bread in the Breadbox | 432 | 12 | 1 |
+| 3 | Reuben and Cherise | 367 | 45 | 2 |
+| 4 | Purple Rain | 183 | 49 | 3 |
+
+**Artists Table**
+
+| id | name |
+| --- | --- |
+| 1 | Talking Heads |
+| 2 | Jerry Garcia Band |
+| 3 | Prince |
+
+<section class="checks-for-understanding">
+### Independent Practice - Students and Modules
+
+Think about the relationship between students and modules (i.e. "Mod 1: Object Oriented Programming", "Mod 2: Web Applications with Ruby").
+
```

Comment:
Shall we rename these? Or perhaps it doesn't matter.

Reply:
Yea, that's probably a good idea to rename them!